lib/src/main/java/graphql/nadel/engine/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -56,30 +63,37 @@ internal object NadelHydrationFieldsBuilder {
     }
 
     fun makeBatchBackingQueries(
+        executionHints: NadelExecutionHints,
         executionBlueprint: NadelOverallExecutionBlueprint,
         instruction: NadelBatchHydrationFieldInstruction,
         aliasHelper: NadelAliasHelper,
         virtualField: ExecutableNormalizedField,
         argBatches: List<Map<NadelHydrationArgument, NormalizedInputValue>>,
     ): List<ExecutableNormalizedField> {
-        val backingFieldOverallObjectTypeNames = getBackingFieldOverallObjectTypenames(instruction, executionBlueprint)
-        val fieldChildren = deepClone(fields = virtualField.children)
-            .mapNotNull { childField ->
-                val objectTypesAreNotReturnedByBackingField =
-                    backingFieldOverallObjectTypeNames.none { it in childField.objectTypeNames }
+        val fieldChildren = if (executionHints.hydrationFilterObjectTypes()) {
+            deepClone(fields = filterChildren(instruction, virtualField.children)) +
+                makeObjectIdFields(executionBlueprint, aliasHelper, instruction)
+        } else {
```

Comment:
So thing is, we applied this type name filtering for batch hydrations but not for non-batched hydration.

I unified the behavior and also moved the `getBackingFieldOverallObjectTypenames` to be precomputed in the instruction.

Also moved the filtering to before `deepClone` so we don't unnecessarily clone objects we were about to filter anyway.

FF is here to ensure smooth rollout.